backend/server/routers/model.py

```diff
@@ -39,8 +40,19 @@ class CourseDetails(BaseModel):
     is_multiterm: bool
 
 
+class ContainerContent(TypedDict):
+    UOC: NotRequired[int]
+    courses: NotRequired[dict[str, str | list[str]]]
+    type: NotRequired[str]
+
+class Container(TypedDict):
+    name: str
+    content: dict[str, ContainerContent]
+
+StructureType = dict[str, Container]
+
 class Structure(BaseModel):  # this is a copout - we should avoid this
```

Comment:
can remove the `this is a copout` comment now?

Reply:
hmm, i thought i did

Reply:
hmm ok gh is wierd, after i pushed again my diff was corrected lol